tests/unit/help_test.sh

```diff
@@ -32,4 +32,47 @@ function test_kworkflow_man()
   assertEquals "($LINENO) We expected an error message." "$expect" "$output"
 }
 
+# Testing kw --version makes only sense when kw is installed, which is left  for
+# the integration tests. But we can still test kw --version in repomode  because
+# it does not require installation. This is what this test does.
+function test_kworkflow_version_in_repomode()
+{
+  local branch_name
+  local head_hash
+  local output
+  local path_to_git_repository
+  local path_to_kw_repository
+
+  # Initialize variables.
+  branch_name='foo'
+  path_to_git_repository="${SHUNIT_TMPDIR}/git_repo"
+  path_to_kw_repository=$(realpath "$(dirname "${BASH_SOURCE[0]}")"/../..)
+
+  # Copy the local kw to our temporary git repo.
+  cp --recursive "${path_to_kw_repository}" "${path_to_git_repository}"
+
+  # cd into the repo, so we can run git commands there.
+  cd "${path_to_git_repository}" || fail 'Failed to switch to git dir.'
+
+  # Create a new branch (but delete it first if it exists)
+  git branch --quiet -D "${branch_name}" # the -D flag has no long-option equivalent
+  git switch --quiet --create "${branch_name}"
+
+  # Make fake git commits from the created branch.
+  mk_fake_git
+
+  # Get commit hash.
+  head_hash=$(git rev-parse --short HEAD)
+
+  # Get kw --version using KW executable from the git repo.
+  output=$(./kw --version 2> /dev/null)
```

Comment:
By the comment describing this unit test, I see what you are trying to test here, but it seems to me that you are actually doing an integration test. As this is a unit test, I think you should be testing the behavior only of `kworkflow_version`.

The test will probably be a little easier to do as you won't need to copy the whole kw repository, run `./kw --version` directly, can drop the `2> /dev/null`, etc. and would only need to mock the value of `KW_REPO_MODE`, the file `${KW_LIB_DIR}/VERSION` and manipulate a generic git repo (it doesn't need to be a kw one, I think). You can also take the opportunity to test the default behavior of `kworkflow_version`, i.e., the one where `"$KW_REPO_MODE" != 'y'`.

Maybe this is an issue on my end as I have `pre-commit` configured, but, nevertheless, the test should not output these messages (see below).

![Screenshot from 2023-12-15 13-24-53](https://github.com/kworkflow/kworkflow/assets/95055784/17595d76-938b-4f25-89e4-37fdf95adabd)